crates/networking/p2p/sync/metrics.rs

```diff
@@ -0,0 +1,180 @@
+            current_cycle: ExecutionCycle {
+                started_at_block_num: start_block_num,
+                started_at_block_hash: start_block_hash,
+                ..Default::default()
+            },
+        }
+    }
+
+    pub fn log_cycle(
+        &mut self,
+        executed_blocks: u32,
+        block_num: u64,
+        block_hash: H256,
+        add_block_time: u128,
+        throughput: f64,
+    ) {
+        self.current_cycle.executed_blocks_count += executed_blocks;
+        self.current_cycle.throughput += throughput;
+        self.current_cycle.add_block_time += add_block_time;
+
+        if self.current_cycle.executed_blocks_count >= self.blocks_to_restart_cycle {
+            self.current_cycle.finished_at = Instant::now();
+            self.current_cycle.finished_at_block_num = block_num;
+            self.current_cycle.finished_at_block_hash = block_hash;
+            self.show_stats();
+
+            // restart cycle
+            self.prev_cycle = self.current_cycle.clone();
+            self.current_cycle = ExecutionCycle {
+                started_at_block_num: block_num,
+                started_at_block_hash: block_hash,
+                ..ExecutionCycle::default()
+            };
+        }
+    }
+
+    fn show_stats(&self) {
+        let elapsed = self
+            .current_cycle
+            .finished_at
+            .duration_since(self.current_cycle.started_at)
+            .as_secs();
+
+        let avg = elapsed as f64 / self.current_cycle.executed_blocks_count as f64;
+
+        let prev_elapsed = self
+            .prev_cycle
+            .finished_at
+            .duration_since(self.prev_cycle.started_at)
+            .as_secs();
+
+        let elapsed_diff = elapsed as i128 - prev_elapsed as i128;
+
+        let add_block_time_in_secs = self.current_cycle.add_block_time / 1000;
+        let add_block_time_ratio = add_block_time_in_secs as f64 / elapsed as f64;
+        let blocks_per_second_over_total_time =
+            self.current_cycle.executed_blocks_count as f64 / elapsed as f64;
+        let blocks_per_second_over_add_block =
+            self.current_cycle.executed_blocks_count as f64 / add_block_time_in_secs as f64;
+        let throughput =
+            self.current_cycle.throughput / self.current_cycle.executed_blocks_count as f64;
+
+        tracing::info!(
+            "[SYNCING PERF] Last {} blocks performance:\n\
+        \tTotal time (takes into account network requests): {} seconds\n\
+        \tTime spent adding blocks: {} seconds ~= {:.2}% of total time\n\
+        \tAverage block in total time: {:.3} seconds\n\
+        \tBlocks per second over total time: {:.3}\n\
+        \tBlocks per second add block: {:.3}\n\
+        \tThroughput: {:.3} Gigagas/s\n\
+        \tStarted at block: {} (hash: {:?})\n\
+        \tFinished at block: {} (hash: {:?})\n\
+        \tExecution count: {}\n\
+        \t======= Overall, this cycle took {} seconds with respect to the previous one =======",
```

Comment:
This is not clear, is this the time this cycle took? is it the time difference between this cycle and the previous one? the wording is confusing